### PR TITLE
Index of string now returns string instead of rune

### PR DESCRIPTION
**Pull request by MichaelS11**

### CODE REVIEW

**Comment by mattn** [comment]
You will add more test to check type of `str[0]`?

**Comment by MichaelS11** [comment]
Added a couple more tests. Was that what you wanted? If it does not, not clear what is wanted.

**Comment by MichaelS11** [comment]
Added interface type
Added TestJson
Added tests to TestKindOf

**Comment by MichaelS11** [comment]
@mattn ,
I think the tests provided should work to make sure a string is returned. Are you alright with them?

**Comment by MichaelS11** [comment]
@mattn ,
Is this good?

File: builtins/core.go [hunk 1]
```diff
@@ -300,10 +300,11 @@ func Import(env *vm.Env) *vm.Env {
...
 	})
 
-	env.DefineType("int64", int64(0))
-	env.DefineType("float64", float64(0.0))
+	env.DefineType("interface", reflect.ValueOf([]interface{}{int64(1)}).Index(0).Type())
 	env.DefineType("bool", true)
-	env.DefineType("string", "")
+	env.DefineType("int64", int64(1))
+	env.DefineType("float64", float64(1))
+	env.DefineType("string", "a")
```
Comment (mattn) [inline builtins/core.go]: you will add rune?

**Comment by MichaelS11** [comment]
Added rune

**Comment by mattn** [comment]
Thanks so much

**Pull request merged**